tests/test_features.py

```diff
@@ -74,6 +75,27 @@ def test_upgradeable_comments() -> None:
     assert v1.upgradeable_version == "version_1"
 
 
+def test_contract_comments() -> None:
+    comments = " @title Test Contract\n @dev Test comment"
+
+    solc_select.switch_global_version("0.8.10", always_install=True)
+    slither = Slither("./tests/custom_comments/contract_comment.sol")
+    compilation_unit = slither.compilation_units[0]
+    contract = compilation_unit.get_contract_from_name("A")[0]
+
+    assert contract.comments == comments
+
+    # Old solc versions have a different parsing of comments
+    # the initial space (after *) is also not kept on every line
+    comments = "@title Test Contract\n@dev Test comment"
+    solc_select.switch_global_version("0.5.16", always_install=True)
+    slither = Slither("./tests/custom_comments/contract_comment.sol")
```

Comment:
Is this meant to test the compact AST or should `solc_force_legacy_json` be used?